assets/js/modules/analytics/datastore/properties.js

```diff
@@ -173,21 +173,28 @@ const baseActions = {
 
 			registry.dispatch( STORE_NAME ).setInternalWebPropertyID( internalPropertyID || '' );
 
+			const existingProfileID = registry.select( STORE_NAME ).getProfileID(); // eslint-disable-line @wordpress/no-unused-vars-before-return
+
 			// Clear any profile ID selection in the case that selection falls to the getProfiles resolver.
 			registry.dispatch( STORE_NAME ).setProfileID( '' );
 
 			const profiles = registry.select( STORE_NAME ).getProfiles( accountID, propertyID );
-			if ( property.defaultProfileId && profiles?.some( ( profile ) => profile.id === property.defaultProfileId ) ) { // eslint-disable-line sitekit/acronym-case
-				registry.dispatch( STORE_NAME ).setProfileID( property.defaultProfileId ); // eslint-disable-line sitekit/acronym-case
-				return;
-			}
 
 			if ( profiles === undefined ) {
 				return; // Selection will happen in in getProfiles resolver.
 			}
```

Comment:
If `profiles` is `undefined` and the selected profile was cleared, then this will still cause the same problem as before (selecting the first even when the selected profile belongs to the selected property).

In this case, I think we shouldn't rely on the `getProfiles` resolver to make any selection for us and it should be handled in this action instead.

Here's what I'm thinking:

- Remove the clearing of the profile ID – the `getProfiles` resolver will only set it if it isn't set
- Select `getProfiles` using `__experimentalResolveSelect` and use `commonActions.await` for the result – after which its resolver will be completed so we won't need to worry about it
- Check if the selected profile ID exists in the returned profiles
    - If it does, nothing more should be needed
    - If not, fallback to the default profile ID from the selected property as we do now
    - Finally, fall back to the first profile returned, or if none: profile_create

Reply:
Updated in https://github.com/google/site-kit-wp/pull/3180/commits/3c3c7835690e44b8f0c3e4d81fe5d3014d7a50f7#diff-5da51f519c7f1141354d71a4f94d41b1de0ca5589b8f731be7edaedb1079864dR177